src/main/java/org/opensearch/search/relevance/transformer/kendraintelligentranking/KendraIntelligentRanker.java

```diff
@@ -70,19 +76,44 @@ public List<Setting<?>> getTransformerSettings() {
    * @return boolean decision on whether to re-rank
    */
   @Override
-  public boolean shouldTransform(SearchRequest request, ResultTransformerConfiguration configuration) {
+  public boolean shouldTransform(final SearchRequest request, final ResultTransformerConfiguration configuration) {
     if (request.source() == null) {
       return false;
     }
+    KendraIntelligentRankingConfiguration kendraConfiguration = (KendraIntelligentRankingConfiguration) configuration;
 
-    // Skip if there is scroll, pagination, or sorting.
-    if (request.scroll() != null || request.source().from() > 0 ||
-        (request.source().sorts() != null && !request.source().sorts().isEmpty())) {
+    // Skip if there is scroll, sorting, or the start of the page is greater than the document limit for Kendra Ranking
+    if (request.scroll() != null ||
+        (request.source().sorts() != null && !request.source().sorts().isEmpty()) ||
+        request.source().from() >= kendraConfiguration.getProperties().getDocLimit()) {
       return false;
     }
     return true;
   }
 
+  @Override
+  public SearchRequest preprocessRequest(final SearchRequest request,
+      final SearchRequest originalSearchRequest,
+      final ResultTransformerConfiguration configuration) {
+    // Source is returned in response hits by default. If disabled by the user, overwrite and enable
+    // in order to access document contents for reranking, then suppress at response time.
+    if (request.source().fetchSource() != null && !request.source().fetchSource().fetchSource()) {
+      originalSearchRequest.source().fetchSource(request.source().fetchSource());
```

Comment:
(Responding to 3 of your comments)

Good catch, this is is not a very clean solution, and I can probably name variables better. The limitation is that the `SearchRequest` class in OpenSearch doesn’t have a copy constructor (I found an [issue](https://github.com/opensearch-project/OpenSearch/issues/869) requesting one, but work doesn’t seem to have been started yet). I spent 2-3 hours trying to create one, but it wasn’t straightforward.

Problem: If we preprocess the request (e.g. to override `_source: false` or the number of documents fetched), we need a mechanism to reverse this at response time.

Ideal Solution: 
- Make a copy of the SearchRequest as it was received
- Preprocess to override params
- Run through OpenSearch dataflow with the request with modified params
- Post-process using the copy so that the customer response is as expect

Since we can’t make a true copy
- I’m creating a basic SearchRequest object with only indices set
- Whatever attributes I modify in `preprocessRequest`, I only update the original values in the “originalRequest” - I didn’t want to name it `searchRequestCopy`  because it’s not a true copy - open to other suggestions, though.
- I use the “originalRequest” to do the post-processing

I thought this was better than passing individual variables, or a map or some sort. And given that the copy constructor issue seems to be getting some attention, I thought we could update the code sometime soon to use that.

Let me know if you have other suggestions.

Reply:
If we can't make a copy, then I think I prefer passing variables. Calling it a originalSearchRequest is super confusing and it seems like we only use a few fields in "originalSearchRequest" (fetchSource, from, and size). I think putting these three fields in a custom object will be better.